### DISCUSSION TITLE
Tabs app using python

### DISCUSSION THREAD

**Discussion by Atharva-Malode** [microsoft/hack-together-teams#30] [Q&A]
Hello developers, just need a help do we have any documentation ready for making a teams tab app using python (flask). I am unable to find resources regarding this. Can anyone help me out?

**Comment by BobGerman** [comment]
Hi @Atharva-Malode - sorry we don't have specific docs or samples for Python - but please be aware that any web page that runs in an IFrame can be a tab, so it's not an issue to write tab apps in Python.

There is a set of labs and related material at https://aka.ms/app-camp that shows how to build an app with NO special tooling. It's unfortunately not Python - it's vanilla JavaScript/Express - but the idea is to teach the principles so a developer can use any stack that is capable of serving up web pages.

Hoping that helps - thanks!

**Comment by svrooij** [comment]
@Atharva-Malode If your python outputs a web page you can turn it into an app. You just have to call a simple JavaScript once the page is loaded to tell teams your page is in fact a teams tab.

Be aware all the integrations (single-sign-on, access token, user info,teams context) are JavaScript based. As in you have to call a JavaScript function to get the user context or info from teams.
There is nothing holding you back to call those JavaScript features from your Python generated html.

On the other hand, if you want a click and debug experience, you will have to go with the provided react or Blazor templates.
